### PR TITLE
Add Redis guides

### PR DESCRIPTION
**Pull request by sebasoga**
Thought it would be a good idea to document some guidelines we should all keep in mind.

### CODE REVIEW

**Comment by cintamani** [comment]
Shall we open a RFC issue and start talking about how we want to use Redis? 

For example, if Redis is used as a cache, we want to make sure that: 
 - All keys are expired and renewed at request 
 - Redis can go down without impacting user experience 

If redis is used as a store, we want to make sure: 
 - We analyse in advance the memory usage of the instance

**Comment by sebasoga** [comment]
@cintamani that sounds like a good idea! A way of doing that could be by creating another adding more guidelines to the `redis/README.md` file this PR introduces.

File: redis/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Redis
+
+- <a name="dont-use-keys-in-production"></a>
+  Redis is threaded this means some commands can potentially block the instance while being
```
Comment (errm) [inline redis/README.md]: single threaded

File: redis/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Redis
+
+- <a name="dont-use-keys-in-production"></a>
+  Redis is threaded this means some commands can potentially block the instance while being 
+  executed, such is the case for `KEYS` (`redis.keys(*)`), look for alternative operation 
+  like `SCAN`. When in doubt look at [Redis' oficial docs](https://redis.io/commands) or
```
Comment (l15n) [inline redis/README.md]: [nit: spelling] `official`

File: redis/README.md (hunk 1 above)
Comment (balvig) [inline redis/README.md]: Can we shorten this to a one-line, a la the other guides? Ie basically "Don't use `keys` in production". Then perhaps it's enough to keep the "why" in the PR?

File: redis/README.md (hunk 2 above)
Comment (cintamani) [reply redis/README.md]: I would mention tha `scan` is a very slow command ( O(N) with N being the number of keys in the instance). We don't want any code to rely on a scan, just one-off or background job tasks

**Comment by sebasoga** [comment]
Please ♻️
